common/build/eslint-plugin-fluid/src/test/mockFiles/no-unchecked-record-access/fileWithOnlyRecordAccess.ts

```diff
@@ -0,0 +1,78 @@
+const aExpectingStringFromStaticType: string = someObjWithStaticType.a; // ok: Assigning string property to a strict string variable
+const aExpectingStringOrUndefinedFromStaticType: string | undefined = someObjWithStaticType.a; // ok: Assigning string property to a string or undefined variable
+
+const indexedRecordOfStrings: IndexSignatureType = someObjWithStaticType;
+indexedRecordOfStrings.a; // ok: Accessing index property 'a' without requiring a particular result
+indexedRecordOfStrings.a.length; // defect: Accessing length of index property 'a', but 'a' might not be present
+indexedRecordOfStrings["a"]; // ok: Accessing property 'a' using bracket notation
+indexedRecordOfStrings["a"].length; // defect: Accessing length of index property 'a' using bracket notation, but 'a' might not be present
+indexedRecordOfStrings[a].length; // defect: Accessing length of index property 'a' using bracket notation, but 'a' might not be present
+indexedRecordOfStrings[b].length; // defect: Accessing length of index property 'b' using bracket notation, but 'b' might not be present
+indexedRecordOfStrings.a?.length; // ok: Using optional chaining to access length safely handles 'undefined'
+indexedRecordOfStrings.a!.length; // ok: The author says they understand the question raised by check and acknowledge that they have other information expecting that it is actually defined or that they are okay with an exception being raise here if "a" is not present and defined
+indexedRecordOfStrings["a"]?.length; // ok: Using optional chaining to access length using bracket notation safely handles 'undefined'
+indexedRecordOfStrings["a"]!.length; // ok: The author says they understand the question raised by check and acknowledge that they have other information expecting that it is actually defined or that they are okay with an exception being raise here if "a" is not present and defined
+
+if (indexedRecordOfStrings.a) {
+	indexedRecordOfStrings.a.length; // ok: Within a truthy check, 'a' is guaranteed to be defined
+}
+
+if ("a" in indexedRecordOfStrings) {
+	indexedRecordOfStrings.a.length; // ok: Accessing length of property inside an 'in' check, 'a' is guaranteed to be defined
+}
+
+for (const [key, value] of Object.entries(indexedRecordOfStrings)) {
+	value.length; // ok: Object.entries provides only present values
+	indexedRecordOfStrings[key]; // ok: Accessing property while looping though records which acts like a `has` property check
+	indexedRecordOfStrings[key].length; // ok: Accessing property while looping though records which acts like a `has` property check
+}
+
+const aExpectingString: string = indexedRecordOfStrings.a; // defect: Assigning index property 'a' to a strict string variable, but 'a' might not be present
+const aExpectingStringOrUndefined: string | undefined = indexedRecordOfStrings.a; // ok: Assigning index property 'a' to string or undefined variable, 'a' might not be present
```

Comment:
In addition to the variable assignment cases, make sure there are other assign/read like contexts that are caught.
Examples: passing as function argument and assigning to property in another object.